apps/system/js/app_usage_metrics.js

```diff
@@ -671,20 +728,35 @@
     this.relativeStartTime = performance.now();
   }
 
-  UsageData.prototype.getAppUsage = function(app) {
+  /*
+   * Get app usage for the current date
+   */
+  UsageData.prototype.getAppUsage = function(app, dayKey) {
     var usage = this.data.apps[app];
+    dayKey = dayKey || this.getDayKey();
+
+    // We lazily initialize both the per-app and per-day usage maps
     if (!usage) {
-      // If no usage exists for this app, create a new empty object for it.
-      usage = {
+      this.data.apps[app] = usage = {};
+    }
+
+    var dayUsage = usage[dayKey];
+    if (!dayUsage) {
+      dayUsage = usage[dayKey] = {
         usageTime: 0,
         invocations: 0,
         installs: 0,
         uninstalls: 0,
         activities: {}
       };
-      this.data.apps[app] = usage;
     }
-    return usage;
+    return dayUsage;
+  };
```

Comment:
I had done that originally, but I realized that this way of organizing data decreases the amount of duplicate data in the payload, so instead of seeing..

```json
{"apps": {
  "20150202":  {
    "A_REALLY_LONG_APPURL": {}
  },
  "20150203": {
    "A_REALLY_LONG_APP_URL": {}
  }
}
```

we'd see 
```json
{"apps": {
  "A_REALLY_LONG_APPURL": {
    "20150202": {},
    "20150203": {}
  }
}
```